File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        map ={}
        for i in range(len(arr)):
            map[arr[i]] = map.get(arr[i],0)+1
        arr = []
        for key,value in map.items():
            arr.append([key,value])
        sortedArr = sorted(arr,key=lambda x:(-x[1]))
        size = len(sortedArr)
        for i in range(size):
            if(k >= sortedArr[-1][1]):
                k -= sortedArr[-1][1]
                sortedArr.pop()
        return len(sortedArr)

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findLeastNumOfUniqueInts = function(arr, k) {
    const map = new Map();
    for(let i=0; i<arr.length; i++){
        map.set(arr[i], (map.get(arr[i]) || 0)+1);
    }
    const sortedArr =[];
    for(let [key,value] of map.entries()){
        sortedArr.push([key,value]);
    }
    sortedArr.sort((a,b)=>b[1]-a[1])
    const len = sortedArr.length;
    for(let i=0; i<len; i++){
        if(sortedArr[sortedArr.length-1][1] <= k){
            k -= sortedArr[sortedArr.length-1][1];
            sortedArr.pop()
        }
    }
    return sortedArr.length
};